server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -355,6 +357,7 @@ Runnable getGlobalCheckpointSyncer() {
      On source remote node , it will be REMOTE_MIGRATING_UNSEEDED when relocating from docrep node
      */
     private final ShardMigrationState shardMigrationState;
+    private DiscoveryNodes discoveryNodes;
```

Comment:
will this always have latest nodes ?

Reply:
This will. We update this object on `createShard` as well as `updateShardState`